### DISCUSSION TITLE
Is it possible to displays live information in h2o apps like what we have in scripts?

### DISCUSSION THREAD

**Discussion by mortezaomidi24** [h2oai/wave#1076] [Q&A] [answered]
Hi 
for my recent project I need to write a dashboard that is both interactive and shows live information(update a table every second) at the same time , is there any way to do it ? 
thanks

**Comment by mturoci** [comment, marked as answer]
Hi @mortezaomidi24.

The mechanism for periodic data updates is same for apps and scripts. Our `dashboard` example uses a simple infinite loop with `time.sleep`. The same can be used for Wave apps, but in a real-world scenario you might want to run the periodic part in a separate python thread instead.

**Reply by mortezaomidi24** [reply]
Thank you so much , 
I got the idea but couldn't write a proper code for it, is there any example code to do it?
Like `dashboard` example,  i can write a loop for updating data in a script but in app it didn't work for me

**Reply by mturoci** [reply]
Have a look at our [site_async](https://github.com/h2oai/wave/blob/master/py/examples/site_async.py) example.

**Reply by mortezaomidi24** [reply]
my problem solved 
thank you , again

**Reply by AWRDev** [reply]
Do we need to use a separate python thread because with a single thread ```time.sleep``` will hang the application?

**Reply by mturoci** [reply]
Yes, either threads, processes or asyncio. See the [docs](https://wave.h2o.ai/docs/background/) and this detailed [blog post](https://medium.com/@unusualcode/background-jobs-in-wave-or-how-not-to-kill-your-ui-ae1fed95693a).
